packages/components/src/List/Toolbar/ColumnChooserButton/service/columnChooser.utils.js

```diff
@@ -0,0 +1,43 @@
+function findMatchingIemLabel(label) {
+	return function find(item) {
+		return item.label === label;
+	};
+}
+
+function transformCollection(columnChooserCollection) {
+	return function transform(item) {
+		const itemToMerged = columnChooserCollection.find(findMatchingIemLabel(item.label));
+		if (itemToMerged) {
+			return {
+				...item,
+				...itemToMerged,
+			};
+		}
+		return item;
+	};
+}
+
+/**
+ * Merged the columns chooser collection into the columns list.
+ * @param {array} originalCollection
+ * @param {arrayOf(ColumnChooserPropTypes)} columnChooserCollection
+ */
+export function mergedColumnsChooser(originalCollection, columnChooserCollection = []) {
+	if (columnChooserCollection.length > 0) {
+		return originalCollection.map(transformCollection(columnChooserCollection));
+	}
+	return originalCollection;
+}
+
+export function compareOrder(a, b) {
```

Comment:
I don't understand the logic here.
What I understand:
- you can have an `order` property in your column definition
- if you have it, it will be set before the ones that don't have it
- if you don't have it, it will be placed after
- there is no sorting between the elements that don't have the order
- there is an asc sort for those that have the order

Can you explain this behavior please ?

Reply:
It's the same ordering used in the ListToVirtualizedListOrder so I used the same to keep thing constans. Don't know about all the business rules. 
What i can say is that order is added to the column inside column chooser if they are not present. Also collection from projects are unorderered so we need to order them. I really don't in the mass the pr comment if you'll see this. I copy paste in slack.